
### Ciência de Dados
### prof. Neylson Crepald
### Exercício Avaliativo de Regressão Linear
### Gilmar Penido de Souza


In [5]:
import pandas as pd
import statsmodels.formula.api as sms
import matplotlib.pyplot as plt
import numpy as np
%matplotlib inline

In [17]:
boston = pd.read_csv("https://raw.githubusercontent.com/selva86/datasets/master/BostonHousing.csv")
boston.tail()

crim   zn  indus  chas    nox     rm   age     dis  rad  tax  ptratio  \
501  0.06263  0.0  11.93     0  0.573  6.593  69.1  2.4786    1  273     21.0   
502  0.04527  0.0  11.93     0  0.573  6.120  76.7  2.2875    1  273     21.0   
503  0.06076  0.0  11.93     0  0.573  6.976  91.0  2.1675    1  273     21.0   
504  0.10959  0.0  11.93     0  0.573  6.794  89.3  2.3889    1  273     21.0   
505  0.04741  0.0  11.93     0  0.573  6.030  80.8  2.5050    1  273     21.0   

          b  lstat  medv  
501  391.99   9.67  22.4  
502  396.90   9.08  20.6  
503  396.90   5.64  23.9  
504  393.45   6.48  22.0  
505  396.90   7.88  11.9

### prevendo medv usando como preditor lstat

In [18]:

modelo1 = sms.ols('medv ~ lstat', data = boston).fit()
print(modelo_medv.summary())

                            OLS Regression Results                            
Dep. Variable:                   medv   R-squared:                       0.544
Model:                            OLS   Adj. R-squared:                  0.543
Method:                 Least Squares   F-statistic:                     601.6
Date:                Thu, 14 Mar 2019   Prob (F-statistic):           5.08e-88
Time:                        23:15:36   Log-Likelihood:                -1641.5
No. Observations:                 506   AIC:                             3287.
Df Residuals:                     504   BIC:                             3295.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     34.5538      0.563     61.415      0.0

### prevendo medv usando como preditor age

In [16]:
modelo2 = sms.ols("medv ~ age", data = boston).fit()
print(modelo2.summary())

                            OLS Regression Results                            
Dep. Variable:                   medv   R-squared:                       0.142
Model:                            OLS   Adj. R-squared:                  0.140
Method:                 Least Squares   F-statistic:                     83.48
Date:                Thu, 14 Mar 2019   Prob (F-statistic):           1.57e-18
Time:                        23:13:27   Log-Likelihood:                -1801.5
No. Observations:                 506   AIC:                             3607.
Df Residuals:                     504   BIC:                             3615.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     30.9787      0.999     31.006      0.0

### usando ambos os preditores

In [15]:

modelo3 = sms.ols("medv ~ lstat + age", data = boston).fit()
print(modelo3.summary())

                            OLS Regression Results                            
Dep. Variable:                   medv   R-squared:                       0.551
Model:                            OLS   Adj. R-squared:                  0.549
Method:                 Least Squares   F-statistic:                     309.0
Date:                Thu, 14 Mar 2019   Prob (F-statistic):           2.98e-88
Time:                        23:11:30   Log-Likelihood:                -1637.5
No. Observations:                 506   AIC:                             3281.
Df Residuals:                     503   BIC:                             3294.
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     33.2228      0.731     45.458      0.0

### O que podemos dizer sobre o comportamento dos coeficientes estimados? E sobre o ajuste dos modelos?

* lstat tem correlaçao com medv poisr P > 0,05 e R-saquared indica que este modelo explica 54% dos casos, neste modelo, para cada 